File: ibkr_show_frontend/src/types/charts.ts
export interface EquityCurvePoint {
  report_date: string
  total_equity: number | null
  total_pnl: number | null
  net_cost: number | null
  realized_pnl: number | null
  daily_mtm: number | null
  daily_twr: number | null
  cnav_twr: number | null
}

export interface EquityCurveResponse {
  items: EquityCurvePoint[]
}

File: ibkr_show_backend/app/schemas/charts.py
from pydantic import BaseModel


class EquityCurvePoint(BaseModel):
    report_date: str
    total_equity: float | None = None
    total_pnl: float | None = None
    net_cost: float | None = None
    realized_pnl: float | None = None
    daily_mtm: float | None = None
    daily_twr: float | None = None


class EquityCurveResponse(BaseModel):
    items: list[EquityCurvePoint]
